packages/scxa-tsne-plot/src/GeneExpressionTSnePlot.js

```diff
@@ -129,59 +130,78 @@ const GeneExpressionScatterPlot = (props) => {
       }
     },
     colorAxis: plotIsDisabled ?
-      {} :
-      {
-        allowNegativeLog: true,
-        min: 0.1,
-        max: 10 ** dataScale,
-        type: `logarithmic`,
-        stops: adjustedStops.map((e, idx) => [1 / adjustedStops.length * idx, e.color]),
-        marker: {
-          color: `#c4463a`
+        {} :
+        {
+          allowNegativeLog: true,
+          min: 0.1,
+          max: 10 ** dataScale,
+          type: `logarithmic`,
+          stops: adjustedStops.map((e, idx) => [1 / adjustedStops.length * idx, e.color]),
+          marker: {
+            color: `#c4463a`
+          },
+          endOnTick: false
         },
-        endOnTick: false
-      },
     legend: plotIsDisabled ?
-      {
-        enabled: false
-      } :
-      {
-        title: {
-          text: `Expression level (CPM)`
-        },
-        floating: false,
-        align: `center`,
-        symbolHeight: 5,
-        symbolWidth: 450,
-        enabled: true
-      }
+        {
+          enabled: false
+        } :
+        {
+          title: {
+            text: `Expression level (CPM)`
+          },
+          floating: false,
+          align: `center`,
+          symbolHeight: 5,
+          symbolWidth: 450,
+          enabled: true
+        }
+  }
+
+  const geneIdOptions = geneIds.sort((a, b) => a - b).map((geneId) => ({
+    value: geneId,
+    label: geneId
+  }))
+
+  let geneIdControl
```

Comment:
Better to use `const` and the ternary operator to do conditional rendering. There’s nothing essentially wrong with the above, but we use the latter pattern elsewhere on our codebase.